app/code/Magento/Catalog/Test/Unit/Controller/Category/ViewTest.php

```diff
@@ -30,219 +36,239 @@
-        $this->context->expects($this->any())->method('getResultFactory')
-            ->willReturn($this->resultFactory);
+        $this->contextMock = $this->createMock(Context::class);
+        $this->contextMock->method('getRequest')->willReturn($this->requestMock);
+        $this->contextMock->method('getResponse')->willReturn($this->responseMock);
+        $this->contextMock->method('getObjectManager')
+            ->willReturn($this->objectManagerMock);
+        $this->contextMock->method('getEventManager')->willReturn($this->eventManagerMock);
+        $this->contextMock->method('getView')->willReturn($this->viewMock);
+        $this->contextMock->method('getResultFactory')
+            ->willReturn($this->resultFactoryMock);
 
-        $this->category = $this->createMock(\Magento\Catalog\Model\Category::class);
-        $this->categoryRepository = $this->getMockForAbstractClass(CategoryRepositoryInterface::class);
+        $this->categoryMock = $this->createMock(\Magento\Catalog\Model\Category::class);
+        $this->categoryRepositoryMock = $this->getMockForAbstractClass(CategoryRepositoryInterface::class);
 
-        $this->store = $this->createMock(Store::class);
-        $this->storeManager = $this->getMockForAbstractClass(StoreManagerInterface::class);
-        $this->storeManager->expects($this->any())->method('getStore')->willReturn($this->store);
+        $this->storeMock = $this->createMock(Store::class);
+        $this->storeManagerMock = $this->getMockForAbstractClass(StoreManagerInterface::class);
+        $this->storeManagerMock->method('getStore')->willReturn($this->storeMock);
 
-        $this->catalogDesign = $this->createMock(Design::class);
+        $this->catalogDesignMock = $this->createMock(Design::class);
 
-        $resultPageFactory = $this->getMockBuilder(PageFactory::class)
+        $resultPageFactoryMock = $this->getMockBuilder(PageFactory::class)
             ->disableOriginalConstructor()
             ->setMethods(['create'])
             ->getMock();
-        $resultPageFactory->expects($this->atLeastOnce())
+        $resultPageFactoryMock->expects($this->atLeastOnce())
             ->method('create')
-            ->willReturn($this->page);
-
-        $this->action = (new ObjectManager($this))->getObject(
-            View::class,
-            [
-                'context' => $this->context,
-                'catalogDesign' => $this->catalogDesign,
-                'categoryRepository' => $this->categoryRepository,
-                'storeManager' => $this->storeManager,
-                'resultPageFactory' => $resultPageFactory,
-                'categoryHelper' => $this->categoryHelper
-            ]
+            ->willReturn($this->pageMock);
+
+        $this->action = new View(
+            $this->contextMock,
+            $this->catalogDesignMock,
+            $this->createMock(Session::class),
+            new Registry(),
```

Comment:
Could you use mock of registry here? I believe because of that we have failing MHI test

Reply:
Sure, changed that. Let's see now